src/charm.py

```diff
@@ -261,12 +72,37 @@ def _on_collect_status(self, e: CollectStatusEvent):
     @property
     def tempo_endpoint(self) -> Optional[str]:
         """Tempo endpoint for charm tracing."""
-        return self.tempo_cluster.get_tracing_endpoint("otlp_http")
+        if endpoints := self.worker.cluster.get_tracing_receivers():
+            return endpoints.get("otlp_http", None)
 
     @property
     def ca_cert_path(self) -> Optional[str]:
         """CA certificate path for tls tracing."""
-        return TEMPO_CA_FILE if Path(TEMPO_CA_FILE).exists() else None
+        return CLIENT_CA_FILE
+
+    def pebble_layer(self, worker: Worker) -> Layer:
+        """Return a dictionary representing a Pebble layer.
+
+        Caller is responsible for checking whether the tempo role is valid before
+        calling this method.
+        """
+        role = "".join(worker.roles)
+        if role == "all":
+            role = "scalable-single-binary"
+        return Layer(
+            {
+                "summary": "tempo worker layer",
+                "description": "pebble config layer for tempo worker",
+                "services": {
+                    "tempo": {
+                        "override": "replace",
+                        "summary": "tempo worker process",
+                        "command": f"/bin/tempo -config.file={CONFIG_FILE} -target {role}",
+                        "startup": "enabled",
```

Comment:
Iiuc, we need to have some config in place to start tempo correctly, so not sure if "startup enabled" is appropriate.

Reply:
Yes. So, I've added a safeguard in the shared worker to not start/restart the pebble service if the config doesn't exist yet. Once the file gets created, I believe we have a guarantee that during the workload lifecycle, we'll always have that file (no other service deletes it anywhere).